yccp/sweeps/transforms.py

```diff
@@ -168,6 +168,41 @@ def describe(self, paramset):
                     " [taken from {}]".format(self.prms["path_from"]))
 
 
+class ApplyFunction(AddValue):
+    """
+    Apply the path_from value to function and save in path_to
+    """
+    default_parameters = {
+        "path_from": None,
+        "path_to": None,
+        "function": None,
+    }
+
+    def modify(self, paramset):
+        assert callable(self.prms['function'])
+        self.get_orig_value(paramset)
+
+        self.final_value = self.prms['function'](self.orig_value)
+        _u.set_recursive(
+            paramset.data,
+            self.prms["path_to"],
+            self.final_value)
+
+    def describe(self, paramset):
+        """
+            Return an object describing the transformation.
+        """
+        assert callable(self.prms['function'])
+        import inspect
+        return "applied {} to {} and saved at {} ({} -> {}).".format(
+            inspect.getsource(self.prms["function"]),
```

Comment:
This will raise an `IOError` if the source is not availble → should be handled!

Reply:
New commit tries (`try`) to `inspect.getsource` and handles possible exception.